### DISCUSSION TITLE
How do I capture ^C to do cleanup?

### DISCUSSION THREAD

**Discussion by feihong** [janet-lang/janet#879] [Q&A]
When I'm running my program and press ctrl+c to stop it, is there some way to capture that so my program can do a little cleanup? I tried using `defer` for this but that doesn't work if the whole program is terminated.

**Comment by andrewchambers** [comment]
I have a package that helps with this: https://github.com/andrewchambers/janet-ctrl-c - it requires the janet event loop to work and is currently linux only.
